client/src/index.js

```diff
@@ -0,0 +1,72 @@
+import axios from 'axios'
+
+const defineProperty = Object.defineProperty
+
+const isBrowser = typeof window !== 'undefined' && typeof document !== 'undefined'
+
+export class ApiClient {
+  constructor(options = {}) {
+
+    defineProperty(this, 'options', {
+      enumerable: false,
+      value: ({ ...DefaultOptions, ...options })
+    })
+
+    const storage = isBrowser ? window.localStorage : require('./storage')()
+
+    defineProperty(this, 'storage', {
+      enumerable: false,
+      value: storage
+    })
+
+    defineProperty(this, 'api', {
+      enumerable: false,
+      value: axios.create({
+        baseURL: this.baseURL,
+        headers: this.authHeaders
+      })
+    })
+  }
+
+  setDefaults(values = {}) {
+    this.instance.defaults = {
+      ...this.instance.defaults,
+      ...values
+    }
+  }
+
+  get baseURL() {
```

Comment:
I am pleased then to introduce you to one of my favorite es6 language enhancements: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get